### PR TITLE
Completely replace req.body/req.params/etc. with the coerced/validated value from Joi

### PR DESCRIPTION
**Pull request by papandreou**
Hi!

I use `context:true` and have run into some cases where setting `stripUnknown` option didn't prevent unknown properties from "slipping past" the validation. This has been a source of some potential mass-assignment bugs in an application I'm working on. Now I want to make sure that object properties not explicitly listed in my schema don't make it into my request handlers, where it's often a temptation to pass the whole request body on to the application code.

Basically this is the problem described in https://github.com/AndrewKeig/express-validation/issues/116

Previously https://github.com/AndrewKeig/express-validation/pull/46 was partially landed(?) but it doesn't work that way in the latest `express-validation` 3.

I think this is a better fix than https://github.com/AndrewKeig/express-validation/pull/151

### CODE REVIEW

File: __tests__/unit/validate.test.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe('Validate', () => {
       expect(() => {
         const middleware = validate(schema, {}, {});
         middleware(null, {}, () => { });
-      }).toThrow(/Cannot read property 'headers' of null/);
+      }).toThrow(/Cannot read property 'headers' of null|Cannot read properties of null \(reading 'headers'\)/);
```
Comment (papandreou) [inline __tests__/unit/validate.test.js:33]: Drive-by: Needed to get the build passing in newer node.js versions.

**Comment by papandreou** [comment]
I've published a [fork](https://www.npmjs.com/package/express-validation-papandreou) with this fix in it and rolled it out in 3 applications. It caught several instances of unvalidated input being used by the application.

@AndrewKeig, what say you? Would be really nice to get this included in an official release 🤗

**Comment by AndrewKeig** [comment]
so sorry @papandreou; I am super busy, and not best placed to maintain this at the moment, I have added you as a contributor, send me your npm handle and I will give you npm publish access. andrew.keig@gmail.com

**Comment by AndrewKeig** [comment]
@papandreou, if you have time feel free to review some of the other issues and prs. thanks again..

**Comment by papandreou** [comment]
@AndrewKeig, thanks! I'll take a look.
